companion guide: ama questions (#216)
decentralized-identity/decentralized-web-node: docs/companion_guide.md

### Q&A

#### General Questions

- **How do you pronounce DWNs?**: We've heard a few ways to say it:
  - DAWN
  - D Web Node
  - D W N
- **How are DWNs different than SOLID Pods?**

> - **For the base case many DWNs should I expect a particular person to have?** A person is

Reviewer: ```suggestion
- **For the base case, how many DWNs should I expect a particular person to have?** A person is
```

Author: thanks @TallTed !

  expected to have a few DWNs. Maybe more than 1 but less than 10.

#### Security Questions

> - **Are there Data Privacy Considerations like GDPR. And how are they taken care of in this kinds of a paradigm?**
>   DWN's are basically GRPR by default as you are 100% in control of your data store.

Reviewer: ```suggestion
- **Are there Data Privacy Considerations like GDPR? And how are they taken care of in this kind of paradigm?**
  DWN's are basically GDPR by default, as you are 100% in control of your data store.
```

- **What is the best way to ensure that recipients of PII access via DWN are not
  persistently storing the information using their own digital agent?**
  The in-browser embedded DWN does have storage restrictions. Let’s say I have a

>   second DWN hosted by some provider, can the second DWN use a different store,

Reviewer: ```suggestion
  second DWN hosted by some provider, can the second DWN use a different store?
```

#### Specification Questions

- **How flexible/dynamic are the protocol control rules?** The point of

>   protocols is so you can create a protocol. What type of actors can do what. Not

Reviewer: "The point of protocols is so you can create a protocol."

The intended meaning of this circular sentence is indecipherable.

Author: @TallTed absolutely. To give myself some clearance here, these were notes I took on the fly during the AMA call. I won't be moving out of draft until I've cleaned up some of these questions and gotten them more consumable for everyone else. It's mostly a PR for visibility, as we discussed this over the DWN call last week. That being said, I very much appreciate the review and encourage you to continue to review the PRs that come forward. I'll probably make a cleanup round by the next DWN call (next week) where I'll move out of draft into an actual PR. 

Thanks!

- **How does the CRDT system work?** There are 2 levels of CRDT. The base layer,
  object level CRDT, and the second layer, which is the data CRDT. These are
  managed with commit strategies. See [here] for more information.
- **Does a DWN run in the cloud, local, or both?**

>   At the very least, they will probaby run locally, and there is a high

Reviewer: ```suggestion
  At the very least, they will probably run locally, and there is a high
```

  likelihood that they will also run in the cloud. The remote data will be
  available in case it needs to be very available.
- **Do we write into an IPFS vs. IPLD Node?**
  DWN use IPLD as an encoding format, but it's not required to throw out to the
  IPFS layer.
- **What is IPLD?**
- **What if you want to use a DWN and don't want data on IPFS?**

> - **Does DWN allows or foresees applications that need cross user/company
>   synchronisation? E.g., DeFi applications that pose double-spend risks or supply chain applications including international participants to be synced**

Reviewer: ```suggestion
- **Does DWN allow or foresee applications that need cross user/company
  synchronisation (e.g., DeFi applications that pose double-spend risks or supply chain applications including international participants to be synced)? **
```
